pkg/domain/domain.go

```diff
@@ -2045,7 +2051,11 @@ func (do *Domain) LoadPrivilegeLoop(sctx sessionctx.Context) error {
 				}
 			case <-time.After(duration):
 				event.All = true
-				event = batchReadMoreData(watchCh, event)
+				event = do.batchReadMoreData(watchCh, event)
+			}
+
+			if !event.All && len(event.UserList) == 0 {
+				continue
 			}
```

Comment:
How could this happen?

Reply:
OK, I see, you skip decoding the event ...
Why not decode the event faithfully, and avoid the unnecessary reloading in `privReloadEvent` function?